app/views/shared/_nav_branded.html.slim

```diff
@@ -2,7 +2,8 @@ nav.py2.sm-pt3.sm-pb0.bg-light-blue.center
   .my1.sm-mb0.relative
     = image_tag(asset_url('logo.svg'), width: 124,
       alt: APP_NAME, class: 'align-middle')
-    .px2.inline-block
-      span.absolute.top-0.bottom-0.border-right
-    = image_tag(asset_url("sp-logos/#{@sp_logo}"), height: 50,
-      alt: @sp_name, class: 'align-middle')
+    - if @sp_logo
```

Comment:
The problem is that [`decorated_session` is determined based on the presence of either the `agency` or `friendly_name`](https://github.com/18F/identity-idp/blob/master/app/helpers/application_helper.rb#L28-L34), so if an SP has a name but no logo, it will still use the `ServiceProviderSessionDecorator`.